<?xml version="1.0" encoding="utf-8"?>

<launch>

	<arg name="result_display" default="true"/>
	<arg name="result_autosize" default="true"/>
	<arg name="result_draw_markers" default="false" />
	<arg name="result_draw_markers_cube" default="true" />
	<arg name="result_draw_markers_axis" default="false" />

	<arg name="uid" default=""/>
	<arg name="video_namespace" default="/stream_pointgrey"/>
	<arg name="video_image_topic" default="image_rect"/>
	<arg name="video_info_topic" default="camera_info"/>
	<arg name="video_rectified" default="true" />

	<arg name="board_config" default="$(find ar_sys)/data/single/board.yml" />
	<arg name="board_frame_id" default="board1" />
	<arg name="marker_size" default=".034" />
  <arg name="publish_tf" default="false" />
	<arg name="publish_pose" default="true" />
	<arg name="publish_corners" default="true" />
	<arg name="calculate_pose" default="false" />

  <node ns="$(arg video_namespace)" pkg="ar_sys" type="single_board" name="ar_single_board$(arg uid)" output="screen">
    <remap from="/camera_info" to="$(arg video_namespace)/$(arg video_info_topic)" />
		<remap from="/image" to="$(arg video_namespace)/$(arg video_image_topic)" />

		<param name="image_is_rectified" type="bool" value="$(arg video_rectified)"/>
		<param name="board_config" type="string" value="$(arg board_config)"/>
		<param name="board_frame" type="string" value="/$(arg board_frame_id)" />
		<param name="marker_size" type="double" value="$(arg marker_size)"/>
		<param name="draw_markers" type="bool" value="$(arg result_draw_markers)" />
		<param name="calculate_pose" type="bool" value="$(arg calculate_pose)" />
		<param name="draw_markers_cube" type="bool" value="$(arg result_draw_markers_cube)" />
		<param name="draw_markers_axis" type="bool" value="$(arg result_draw_markers_axis)" />
		<param name="publish_pose" type="bool" value="$(arg publish_pose)" />
		<param name="publish_corners" type="bool" value="$(arg publish_corners)" />
        <param name="publish_tf" value="$(arg publish_tf)" />
	</node>

  <group if="$(arg result_display)">
		<node ns="$(arg video_namespace)" pkg="image_view" type="image_view" name="image_view$(arg uid)" output="screen">
			<remap from="image" to="ar_single_board$(arg uid)/result" />

			<param name="autosize" type="bool" value="$(arg result_autosize)" />
		</node>
	</group>

</launch>
